packages/react-urql/src/hooks/useSource.ts

```diff
@@ -19,6 +19,8 @@ import { useClient } from '../context';
 let currentInit = false;
```

Comment:
should we move this inside the hook as well? all be it just to unmark this as a side-effect?

Reply:
It needs to stay outside the hook for cross-component updates to be prevented